backend/src/api/model/series.rs

```diff
@@ -327,6 +349,193 @@ impl Series {
             out
         }).await
     }
+
+    pub(crate) async fn update_acl(id: Id, acl: Vec<AclInputEntry>, context: &Context) -> ApiResult<Series> {
+        if !context.config.general.allow_acl_edit {
+            return Err(err::not_authorized!("editing ACLs is not allowed"));
+        }
+
+        let series = Self::load_for_api(
+            id,
+            context,
+            err::invalid_input!(
+                key = "series.acl.not-found",
+                "series not found",
+            ),
+            err::not_authorized!(
+                key = "series.acl.not-allowed",
+                "acl update not allowed",
+            )
+        ).await?;
+
+        info!(series_id = %id, "Requesting ACL update of series");
+
+        let response = context
+            .oc_client
+            .update_acl(&series, &acl, context)
+            .await
+            .map_err(|e| {
+                error!("Failed to send acl update request: {}", e);
+                err::opencast_unavailable!("Failed to send acl update request")
+            })?;
+
+        if response.status() == StatusCode::OK {
+            // 200: The updated access control list is returned.
+            let db_acl = convert_acl_input(acl);
+
+            context.db.execute("\
+                update series \
+                set read_roles = $2, write_roles = $3 \
+                where id = $1 \
+            ", &[&series.key, &db_acl.read_roles, &db_acl.write_roles]).await?;
+
+            if context.config.general.lock_acl_to_series {
+                context.db.execute("\
+                    update events \
+                    set read_roles = $2, write_roles = $3 \
+                    where series = $1 \
+                ", &[&series.key, &db_acl.read_roles, &db_acl.write_roles]).await?;
+            }
+
+            Self::load_by_id(id, context)
+                .await?
+                .ok_or_else(|| err::invalid_input!(
+                    key = "series.acl.not-found",
+                    "series not found",
+                ))
+        } else {
+            warn!(
+                series_id = %id,
+                "Failed to update series acl, OC returned status: {}",
+                response.status(),
+            );
+            Err(err::opencast_error!("Opencast API error: {}", response.status()))
+        }
+    }
+
+    pub(crate) async fn update_metadata(
+        id: Id,
+        title: &str,
+        description: Option<&str>,
+        context: &Context,
+    ) -> ApiResult<Series> {
+        let series = Self::load_for_api(
+            id,
+            context,
+            err::invalid_input!(
+                key = "series.metadata.not-found",
+                "series not found",
+            ),
```

Comment:
This error is the same as in `delete` and `update_acl`. Maybe we should just hardcode it in `load_for_api`? I can't think of a reason why any caller would ever want to change this. (Same for `event.rs`)

Reply:
It's not the same error. At least they don't use the same message. There are variations depending on the used endpoint (metadata, delete, acl). But I can generalize these.

Reply:
Ah so for the second error (not authorized) I did realize that they are different errors. And my comment didn't suggest making them all the same. I didn't realize the "not found" error was also different per action. Now that you removed all action-specific strings, lets keep it like that. While the "video action" phrase isn't that amazing, the context will always make it very clear what it means. This error message will appear next to the delete button for example. So yeah, I think it's fine and cuts down on translation strings, which is nice.

Reply:
Yeah, the video/series "action" phrasing is meh. I considered sth like "not allowed to *change* this item".. which would also be ok I think? It would also refer to deleting, but that could be considered a "change" I guess.